### PR TITLE
Refactor Scan Pull Request to accept PR ID as input

### PR DESCRIPTION
**Pull request by EyalDelarea**
- [x] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
---

Refactor `scan-pull-request` command to accept PR ID as input instead of relying on context.

- Deleted all the git test folders for scan pull requests tests as we are downloading only specific branches.
- Refactored `scan-pull-requests command` to use the singular `scan-pull-request`
- Update Github templates

This will allow to use scan pull requests command with Jenkins webhook trigger, providing the pull request ID.

Jenkins Webhook Docs Update PR: 
https://github.com/jfrog/frogbot/pull/399

Update:
When pull request origin from a forked repository, we need to know the owner in order to access the source code.
https://github.com/jfrog/froggit-go/pull/102

### CODE REVIEW

File: commands/utils/params.go [hunk 1]
```diff
@@ -232,15 +236,32 @@ func (g *Git) setDefaultsIfNeeded(git *Git) (err error) {
...
+			if err != nil {
+				return err
+			}
+			g.PullRequestID = idNum
 		}
 	}
 	return
 }
 
+func (g *Git) setDefaultApiEndpoint() string {
```
Comment (sverdlov93) [inline commands/utils/params.go:251]: cant we get it from froggit-go?

File: commands/utils/params.go [hunk 2]
```diff
@@ -232,15 +236,32 @@ func (g *Git) setDefaultsIfNeeded(git *Git) (err error) {
...
 	}
-	// Non-mandatory git branch pr id.
-	if pullRequestIDString := getTrimmedEnv(GitPullRequestIDEnv); pullRequestIDString != "" {
-		if g.PullRequestID, err = strconv.Atoi(pullRequestIDString); err != nil {
-			return err
+	if g.PullRequestID == UndefinedPrID {
+		if idStr := getTrimmedEnv(GitPullRequestIDEnv); idStr != "" {
+			idNum, err := strconv.Atoi(idStr)
+			if err != nil {
+				return err
```
Comment (sverdlov93) [inline commands/utils/params.go:243]: add informative error

File: commands/utils/params.go [hunk 3]
```diff
@@ -232,15 +236,32 @@ func (g *Git) setDefaultsIfNeeded(git *Git) (err error) {
 			g.EmailAuthor = frogbotAuthorEmail
 		}
 	}
-	// Non-mandatory git branch pr id.
-	if pullRequestIDString := getTrimmedEnv(GitPullRequestIDEnv); pullRequestIDString != "" {
-		if g.PullRequestID, err = strconv.Atoi(pullRequestIDString); err != nil {
-			return err
+	if g.PullRequestID == UndefinedPrID {
+		if idStr := getTrimmedEnv(GitPullRequestIDEnv); idStr != "" {
```
Comment (sverdlov93) [inline commands/utils/params.go:240]: why PullRequestID can't be string?

**Review by sverdlov93 — COMMENTED** [review]
not finished the review yet

File: commands/utils/params.go (hunk 3 above)
Comment (EyalDelarea) [reply commands/utils/params.go:240]: The API accepts ID as an int ( froggit-go )

File: commands/utils/params.go (hunk 1 above)
Comment (EyalDelarea) [reply commands/utils/params.go]: Moved 👍

File: commands/scanpullrequest.go [hunk 4]
```diff
@@ -98,17 +82,23 @@ func scanPullRequest(repoConfig *utils.Repository, client vcsclient.VcsClient) e
 	return err
 }
 
-func auditPullRequest(repoConfig *utils.Repository, client vcsclient.VcsClient) ([]formats.VulnerabilityOrViolationRow, []formats.IacSecretsRow, error) {
+// Downloads Pull Requests branches code and audits them
+func auditPullRequest(repoConfig *utils.Repository, client vcsclient.VcsClient, pullRequestDetails vcsclient.PullRequestInfo) ([]formats.VulnerabilityOrViolationRow, []formats.IacSecretsRow, error) {
 	var vulnerabilitiesRows []formats.VulnerabilityOrViolationRow
 	var iacRows []formats.IacSecretsRow
-	targetBranch := repoConfig.Branches[0]
+	targetBranch := pullRequestDetails.Target.Name
```
[the remark this answers is not in the document]
Comment (EyalDelarea) [reply commands/scanpullrequest.go:89]: They can be empty with default values, but they cannot be nil.

**Pull request merged**